[nnc] Separate printing of optimized llvm bitcode from assembly

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56119 [nnc] Don't fuse fp16 on CPU
* #56118 [nnc][trivial] Trailing underscore style for llvmCode, asmCode members
* **#56117 [nnc] Separate printing of optimized llvm bitcode from assembly**

I was debugging an issue during instruction selection and wanted to
see the input bitcode.  This way we always print it before going into the asm
generation pass.

Differential Revision: [D27781683](https://our.internmc.facebook.com/intern/diff/D27781683/)

bertmaher committed:
[nnc] Separate printing of optimized llvm bitcode from assembly

I was debugging an issue during instruction selection and wanted to
see the input bitcode.  This way we always print it before going into the asm
generation pass.

Differential Revision: [D27781683](https://our.internmc.facebook.com/intern/diff/D27781683/)

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/llvm_codegen.cpp:
```diff
@@ -550,10 +550,13 @@ void LLVMCodeGenImpl::emitKernel(
   asmBuffer.set_size(0);
   module_->print(asmStream, nullptr);
   llvmCode = asmStream.str().str();
+  GRAPH_DEBUG(
```
Maybe make one of them `GRAPH_UPDATE` to reduce amount of info dumped at once?

ZolotukhinM reviewed (approved):
👍

bertmaher committed:
format on "[nnc] Separate printing of optimized llvm bitcode from assembly"


I was debugging an issue during instruction selection and wanted to
see the input bitcode.  This way we always print it before going into the asm
generation pass.

Differential Revision: [D27781683](https://our.internmc.facebook.com/intern/diff/D27781683/)

[ghstack-poisoned]
